examples/apps/ncp/main.c

```diff
@@ -97,3 +97,24 @@ int main(int argc, char *argv[])
 
     return 0;
 }
+
+
+/*
+ * Provide, if required an "otPlatLog()" function
+ */
+
+#if ((OPENTHREAD_CONFIG_LOG_OUTPUT == OPENTHREAD_CONFIG_LOG_OUTPUT_NONE) ||  (OPENTHREAD_CONFIG_LOG_OUTPUT == OPENTHREAD_CONFIG_LOG_OUTPUT_APP_STREAM))
```

Comment:
Ok... got it...NONE here  means no output at all...

Then I think we should ensure the default setting  is such that the platform developer is expected/required to provide the `otPlatLog()` function... So I think the default value for `OPENTHREAD_CONFIG_LOG_OUTPUT` should be `OPENTHREAD_CONFIG_LOG_OUTPUT_PLATFORM_DEFINED`